### PR TITLE
Add docker-compose fixtures for S3 integration tests

### PR DESCRIPTION
**Pull request by tlrx**
Similarly to what has been done for Azure (#48636) and GCS (#48762), this pull request removes the existing Ant fixture that emulates a S3 storage service in favor of multiple docker-compose based fixtures.

The goals here are multiple:
- be able to reuse a `s3-fixture` outside of the `repository-s3` plugin;
- allow parallel execution of integration tests;
- removes the existing `AmazonS3Fixture` that has evolved in a weird beast in dedicated, more maintainable fixtures.

The server side logic that emulates S3 mostly comes from the latest HttpHandler made for S3 blob store repository tests, with additional features extracted from the (now removed) `AmazonS3Fixture`:
- authentication checks
- session token checks
- improved response errors

I also had to add chunked upload request support for S3 object as it was not fully supported. The server side logic of all tests now reside in a single `S3HttpHandler` class.

Whereas `AmazonS3Fixture` contained logic for basic tests, session token tests, EC2 tests or ECS tests, the S3 fixtures are now dedicated to each kind of test. It's also now to see that fixtures are inheriting from each other, making things easier to maintain or understand (at least to me).

While this change makes things easier to reuse and maintain, I think it has an impact on test execution time but I think it worth it.

### CODE REVIEW

File: plugins/repository-s3/build.gradle [hunk 1]
```diff
@@ -52,7 +52,7 @@ dependencies {
   // and whitelist this hack in JarHell
   compile 'javax.xml.bind:jaxb-api:2.2.2'
 
-  testCompile project(':test:fixtures:minio-fixture')
+  testCompile project(':test:fixtures:s3-fixture')
```
Comment (atorok) [inline plugins/repository-s3/build.gradle]: would be nice to be able to avoid this, but as we discussed in one of the other PRs we can do that separately.

**Review by alpar-t — APPROVED** [review]
LGTM. Thanks!

File: plugins/repository-s3/build.gradle (hunk 1 above)
Comment (mark-vieira) [reply plugins/repository-s3/build.gradle]: What's the concern about having this additional test dependency?

**Review by mark-vieira — APPROVED** [review]
LGTM. Thanks for continuing to replace the usages of `AntFixture`.

**Review by original-brownbear — APPROVED** [review]
LGTM, thanks so much for cleaning this up!

File: plugins/repository-s3/build.gradle (hunk 1 above)
Comment (tlrx) [reply plugins/repository-s3/build.gradle]: @mark-vieira We discussed this point in https://github.com/elastic/elasticsearch/pull/48636#discussion_r340434656.

**Comment by tlrx** [comment]
Thanks @atorok, @mark-vieira and @original-brownbear! I'm happy to see this change merged :)

File: plugins/repository-s3/build.gradle (hunk 1 above)
Comment (mark-vieira) [reply plugins/repository-s3/build.gradle]: I think this is fine.

Comment (tlrx) [reply plugins/repository-s3/build.gradle]: Thanks @mark-vieira for your feedback :+1:

**Pull request merged**